apps/settings/js/panels/call/call_settings.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * Because there are some APIs that are working like mutex, we have to 
+ * wrap them inside a special module and provide a simple interface for
+ * users to use.
+ *
+ * And also, we can use CallSettings.observe('isLocked', ...) to know
+ * what's the current status and do related UI stuffs (like blocking ...)
+ */
+define(function(require) {
+  'use strict';
+
+  var SettingsCache = require('modules/settings_cache');
+  var TaskScheduler = require('modules/task_scheduler');
+  var CallForwarding = require('panels/call/call_forwarding');
+
+  var CallSettings = function() {
+    this._cardIndex = 0;
+    this._conns = window.navigator.mozMobileConnections;
+    this._conn = this._conns[this._cardIndex];
+    this._taskScheduler = TaskScheduler();
+  };
+
+  CallSettings.prototype = {
+    get observe() {
+      // we will pass out the observe out
+      return this._taskScheduler.observe.bind(this._taskScheduler);
+    },
+
+    set cardIndex(cardIndex) {
+      this._cardIndex = cardIndex;
+      this._conn = this._conns[cardIndex];
+    },
+
+    _getConn: function() {
+      return this._conn;
+    },
+
+    getCallWaitingOption: function() {
+      var promise = new Promise((resolve, reject) => {
+        this._taskScheduler.enqueue('CALL_WAITING', (done) => {
+          this._getConn().getCallWaitingOption().then((result) => {
+            done();
+            resolve(result);
+          }, () => {
+            done();  
+            reject();
+          });
+        });
+      });
+      return promise;
+    },
+
+    getCallingLineIdRestriction: function() {
+      var promise = new Promise((resolve, reject) => {
+        this._taskScheduler.enqueue('CALLER_ID_PREF', (done) => {
+          this._getConn().getCallingLineIdRestriction().then((result) => {
+            done();
+            resolve(result);
+          }, () => {
+            done();
+            reject();
+          });
+        });
+      });
+      return promise;
+    },
+
+    getAllOptions: function() {
```

Comment:
The function was originally defined under `CallSettings`. Now it is under `CallForwarding` and it's okay.